docs/framework/setting.md

```diff
@@ -41,35 +42,37 @@ Each file consists of a php snippet which returns an array. Array keys are strin
     'default' => FALSE,
     'html_type' => 'radio',
     'add' => '4.7',
-    'title' => 'Accept profile submissions from external sites',
+    'title' => ts('Accept profile submissions from external sites'),
     'is_domain' => 1,
     'is_contact' => 0,
-    'description' => 'If enabled, CiviCRM will permit submissions from external sites to profiles. This is disabled by default to limit abuse.',
-    'help_text' => NULL,
+    'description' => ts('If enabled, CiviCRM will permit submissions from external sites to profiles. This is disabled by default to limit abuse.'),
   ),
 ```
 
 The Supported Properties for settings are:
 
 | property | Usage | Example Notes |
 | --- | --- | --- |
-| group | ? | Appears to correspond with the name of the file, doesn't that make it redundant? |
-| `group_name` | Name of group this setting belongs to.  These are defined as constants in the class `CRM_Core_BAO_Setting` |  Uses a string & not a constant as it might be defined outside of core too (& the constants are particularly ugly for these). This has been deprecated as of 4.7. Since 4.7, the two main options are `domain` or `contact`. The older names such as `CiviCRM Preferences` are treated as aliases for `domain` or `contact`|
-| name | Name of this setting| This is the same as the array key. Definitely redundant! |
-| type | Data type| String, Array, Integer, Boolean |
-| default | Default value| Value to use if setting is not set. |
-| add | Version added to CiviCRM| |
-| `quick_form_type` | Widget type (admin form)| e.g YesNo, Element  |
-| `html_type` | Html type (admin form)| Used when `quick_form_type` is "Element". |
+| `name` | Name of this setting| This is the same as the array key. Required |
+| `type` | Data type| String, Integer, Boolean (Array is discouraged as current uses of it do not have full handling and String  + a serialize key addresses most uses) |
+| `title` | Title (admin form)| note: use ts() or E::ts() for extensions|
+| `description` | Description (admin form)| note: use ts() or E::ts() for extensions|
+|`serialize`|define how an array of values are stored|ie CRM_Core_DAO::SERIALIZE_JSON is recommended. Other constants exist on the CRM_Core_DAO class|
+| `default` | Default value| Value to use if setting is not set. |
+| `add` | Version added to CiviCRM| |
+| `html_type` | Html type (admin form)| This is the preferred way to describe the html type as it is not quick form specific. It will be used it present. Syntax is lower case. e.g 'select', 'radio', 'checkboxes', 'checkbox', 'text', 'textarea', 'entity_reference|
```

Comment:
@eileenmcnaughton just thinking i note int he DAO metadata we have `<type>Text</type>` or `<type>SelectDate</type>` but here we are going lowercase maybe there is some difference

Reply:
@seamuslee001 Yeah @mattwire was the one who concluded which way to go on casing & TBH I didn't go done the whole rabbit hole - but I think the type in DAO maps to 'quick_form_type' here?